helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -84,27 +88,48 @@ public AccessResult() {
   private static Logger LOG = LoggerFactory.getLogger(ZkBaseDataAccessor.class);
 
   private final HelixZkClient _zkClient;
+  private final HelixZkClient _nonZNRecordClient;
 
   public ZkBaseDataAccessor(HelixZkClient zkClient) {
     if (zkClient == null) {
       throw new NullPointerException("zkclient is null");
     }
     _zkClient = zkClient;
+    _nonZNRecordClient = new ZkClient.Builder().setConnection(
+        new ZkConnection(zkClient.getServers(), ZkClient.DEFAULT_SESSION_TIMEOUT))
+        .setZkSerializer(new ZkSerializer() {
+          @Override
+          public byte[] serialize(Object data) throws ZkMarshallingError {
+            if (data instanceof byte[]) {
+              return (byte[]) data;
+            }
+            throw new HelixException("Only support a byte array as an argument!");
+          }
+
+          @Override
+          public Object deserialize(byte[] data) throws ZkMarshallingError {
+            return data;
+          }
+        }).build();
+  }
+
+  public HelixZkClient getZkClient() {
+    return _zkClient;
   }
```

Comment:
Should we just make _zkClient a protected field instead of creating a public getter?

Reply:
ZkCacheBaseDataAccessor is not a child class of ZkBaseDataAccessor, so making it protected won't help ZkCacheBaseDataAccessor get access to the _zkClient

Reply:
I've used a simple boolean parameter to avoid using the public get method in the latest version.